kv/kvserver: TestFlowControlCrashedNodeV2 failed

kv/kvserver.TestFlowControlCrashedNodeV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/d3a14e91-75ee-4c60-8028-ebbef8963920?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [954761451e68c1c5db5fbe0f336225955dad1075](https://github.com/cockroachdb/cockroach/commits/954761451e68c1c5db5fbe0f336225955dad1075):


```
           range_id, store_id;
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 5.0 MiB               
        +  a        | 2        | 0 B                   
           a        | 3        | 5.0 MiB               
         
         
         -- Flow token metrics from n1 after n2 crashed. Observe that we've returned the
         -- 5MiB previously held by n2.
        --- FAIL: TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_all (7.61s)
=== RUN   TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/crashed_node_v1_encoding_apply_to_elastic:8:
         
        output didn't match expected:
        @@ -71,10 +71,11 @@
           range_id, store_id;
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 5.0 MiB               
        +  a        | 2        | 0 B                   
           a        | 3        | 5.0 MiB               
         
         
         -- Flow token metrics from n1 after n2 crashed. Observe that we've returned the
         -- 5MiB previously held by n2.
        --- FAIL: TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic (8.76s)
=== RUN   TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=2
    --- FAIL: TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=2 (12.35s)
=== RUN   TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/crashed_node_v2_encoding_apply_to_all:1:
         
        output didn't match expected:
        @@ -71,10 +71,11 @@
           range_id, store_id;
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 5.0 MiB               
        +  a        | 2        | 0 B                   
           a        | 3        | 5.0 MiB               
         
         
         -- Flow token metrics from n1 after n2 crashed. Observe that we've returned the
         -- 5MiB previously held by n2.
        --- FAIL: TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (7.18s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlCrashedNodeV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47957

kv/kvserver.TestFlowControlCrashedNodeV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/3af843f7-448d-46d9-9460-f032e8b00bb1?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [954761451e68c1c5db5fbe0f336225955dad1075](https://github.com/cockroachdb/cockroach/commits/954761451e68c1c5db5fbe0f336225955dad1075):


```
=== RUN   TestFlowControlCrashedNodeV2
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFlowControlCrashedNodeV21385738674
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:662: -- test log scope end --
test logs left over in: outputs.zip/logTestFlowControlCrashedNodeV21385738674
--- FAIL: TestFlowControlCrashedNodeV2 (32.38s)
=== RUN   TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=1
    --- FAIL: TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=1 (14.53s)
=== RUN   TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/crashed_node_v1_encoding_apply_to_elastic:8:
         
        output didn't match expected:
        @@ -71,10 +71,11 @@
           range_id, store_id;
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 5.0 MiB               
        +  a        | 2        | 0 B                   
           a        | 3        | 5.0 MiB               
         
         
         -- Flow token metrics from n1 after n2 crashed. Observe that we've returned the
         -- 5MiB previously held by n2.
        --- FAIL: TestFlowControlCrashedNodeV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic (5.61s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlCrashedNodeV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -4411,7 +4411,7 @@ func (h *flowControlTestHelper) waitForConnectedStreams(
 				connected++
 			}
 		}
-		if connected != expConnectedStreams {
+		if len(state.ConnectedStreams) != expConnectedStreams || connected != expConnectedStreams {
 			return fmt.Errorf("expected %d connected streams, got %d/%d",
 				expConnectedStreams, connected, len(state.ConnectedStreams))
 		}

